src/OctoshiftCLI.Tests/OctoLoggerTests.cs

```diff
@@ -24,19 +26,220 @@ public void Secrets_Should_Be_Masked_From_Logs_And_Console()
+            string console = null;
+            string log = null;
+            string verbose = null;
+            string error = null;
+
+            var logger = new OctoLogger(msg => log = msg, msg => verbose = msg, msg => console = msg, msg => error = msg)
+            {
+                Verbose = false
+            };
+
+            const string userFriendlyMessage = "A user friendly message";
+            const string exceptionDetails = "exception details";
+            var octoshiftCliException = new OctoshiftCliException(userFriendlyMessage,
+                new ArgumentNullException("arg", exceptionDetails));
+
+            // Act
+            logger.LogError(octoshiftCliException);
+
+            // Assert
+            console.Should().BeNull();
+
+            error.Trim().Should().EndWith($"[ERROR] {userFriendlyMessage}");
+            error.Should().NotContain(exceptionDetails);
+
+            log.Trim().Should().EndWith($"[ERROR] {userFriendlyMessage}");
+            log.Should().NotContain(exceptionDetails);
+
+            verbose.Trim().Should().EndWith($"[ERROR] {octoshiftCliException}");
+        }
+
+        [Fact]
+        public void LogError_For_Unexpected_Exception_Should_Log_Generic_Error_Message_In_Non_Verbose_Mode()
+        {
+            // Arrange
+            const string genericErrorMessage = "An unexpected error happened. Please see the logs for details.";
+
+            string console = null;
+            string log = null;
+            string verbose = null;
+            string error = null;
+
+            var logger = new OctoLogger(msg => log = msg, msg => verbose = msg, msg => console = msg, msg => error = msg)
+            {
+                Verbose = false
+            };
+
+            const string userEnemyMessage = "Some user enemy error message!";
+            const string exceptionDetails = "exception details";
+            var unexpectedException = new InvalidOperationException(userEnemyMessage,
+                new ArgumentNullException("arg", exceptionDetails));
+
+            // Act
+            logger.LogError(unexpectedException);
+
+            // Assert
+            console.Should().BeNull();
+
+            error.Trim().Should().EndWith($"[ERROR] {genericErrorMessage}");
+            error.Should().NotContain(userEnemyMessage);
+            error.Should().NotContain(exceptionDetails);
+
+            log.Trim().Should().EndWith($"[ERROR] {genericErrorMessage}");
+            log.Should().NotContain(userEnemyMessage);
+            log.Should().NotContain(exceptionDetails);
+
+            verbose.Trim().Should().EndWith($"[ERROR] {unexpectedException}");
+            verbose.Should().NotContain(genericErrorMessage);
+        }
+
+        [Fact]
+        public void LogError_For_Any_Exception_Should_Always_Log_Entire_Exception_In_Verbose_Mode()
+        {
+            // Arrange
+            const string genericErrorMessage = "An unexpeted error happened. Please see the logs for details.";
```

Comment:
```suggestion
            const string genericErrorMessage = "An unexpected error happened. Please see the logs for details.";
```

Reply:
already addressed in 26e2916b1709a4605bfbe578b88edc9f9153ea2f